staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -737,6 +737,9 @@ func deletionFinalizersForGarbageCollection(ctx context.Context, e *Store, acces
 	if !e.EnableGarbageCollection {
 		return false, []string{}
 	}
+	if accessor.GetDeletionTimestamp() != nil {
```

Comment:
I decided not to return an error because the finalizers are added by the system, not by an end user.

Reply:
If the user explicitly set the foregroundDeletion in its deleteOptions, it's bad to omit it silently.